verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_edks/aws_cryptosdk_priv_hdr_parse_edks_harness.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/header.h>
+#include <aws/cryptosdk/private/header.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+/**
+ * In the aws_cryptosdk_enc_ctx_deserilize() proof, the first value we read is the number of elements,
+ * which we need to be constrained in order to ensure that the proof finishes. All other values can be left nondet.
+ * This generates exactly that set of bytes
+ */
+uint16_t aws_byte_cursor_read_be16_generator_for_parse_edks(const struct aws_byte_cursor *cursor) {
+    (void)cursor;
+    static int num_times_called = 0;
+    num_times_called++;
+    uint16_t rval;
+    if (num_times_called == 1) {
+        __CPROVER_assume(rval <= MAX_EDK_LIST_ITEMS);
+    }
+    return rval;
+}
+
+void aws_cryptosdk_priv_hdr_parse_edks_harness() {
+    /* Nondet Input */
+    struct aws_cryptosdk_hdr *hdr   = hdr_setup(MAX_TABLE_SIZE, MAX_EDK_LIST_ITEMS, MAX_BUFFER_SIZE);
+    struct aws_byte_cursor *pcursor = malloc(sizeof(*pcursor));
+
+    /* Assumptions */
+    __CPROVER_assume(&hdr->alloc != NULL);
```

Comment:
You don't need `&hdr->alloc`, right? Why not just `hdr->alloc != NULL`?

Reply:
See above.